libs/vkd3d/memory.c

```diff
@@ -309,28 +345,16 @@ static void vkd3d_memory_transfer_queue_clear_allocation(struct vkd3d_memory_tra
     }
     else if (allocation->resource.vk_buffer)
     {
-        pthread_mutex_lock(&queue->mutex);
-
-        if (!vkd3d_array_reserve((void**)&queue->allocations, &queue->allocations_size,
-                queue->allocations_count + 1, sizeof(*queue->allocations)))
-        {
-            ERR("Failed to insert free range.\n");
-            pthread_mutex_unlock(&queue->mutex);
-            return;
-        }
-
```

Comment:
We're sampling queue->next_signal_value here outside the lock, is that safe?

Reply:
It might be safer to take lock manually then vkd3d_memory_transfer_queue_execute_transfer_locked() instead.

Reply:
Where is that access exactly?

Reply:
Right below here.

```
        allocation->clear_semaphore_value = queue->next_signal_value;

        if (allocation->chunk)
            allocation->chunk->allocation.clear_semaphore_value = queue->next_signal_value;
```

next_signal_value can bump if another thread flushes the clear queue I think?

Reply:
nvm github just made this hard to find.